helix-core/src/test/java/org/apache/helix/integration/task/TestStopWorkflowWithExecutionDelay.java

```diff
@@ -0,0 +1,93 @@
+package org.apache.helix.integration.task;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.helix.TestHelper;
+import org.apache.helix.task.JobConfig;
+import org.apache.helix.task.JobContext;
+import org.apache.helix.task.TaskState;
+import org.apache.helix.task.TaskUtil;
+import org.apache.helix.task.Workflow;
+import org.apache.helix.task.WorkflowContext;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * Test To check if the Execution Delay is respected.
+ */
+public class TestStopWorkflowWithExecutionDelay extends TaskTestBase {
+
+  @BeforeClass
+  public void beforeClass() throws Exception {
+    super.beforeClass();
+  }
+
+  @Test
+  public void testStopWorkflowWithExecutionDelay() throws Exception {
+    // Execution Delay is set to be 20 milliseconds.
+    final long executionDelay = 20L;
```

Comment:
Is there a reason why this is set at 20L? Would it make a difference if it was 200, 2000? Or does any delay do it? Just curious. Might be good to clarify here.

Reply:
No there is no reason to use 20L. As long as we use any execution delay and the job goes to Inflight job queue, we need this fix. I will clarify on the code.